specs/Menu.md

```diff
@@ -1,731 +1,545 @@
-   <li tabindex="-1" role="menuitem" aria-disabled="false">Item three<span class="MuiTouchRipple-root"></span></li>
-   <li tabindex="-1" role="menuitem" aria-disabled="false">Subitem one<span class="MuiTouchRipple-root"></span></li>
-   <li tabindex="-1" role="menuitem" aria-disabled="false">Subitem Two<span class="MuiTouchRipple-root"></span></li>
-</ul>
-```
+```typescript
+// v7 controlled ContextualMenu
+const [showContextualMenu, setShowContextualMenu] = React.useState(false);
+const onShowContextualMenu = () => setShowContextualMenu(true);
+const onHideContextualMenu = () => setShowContextualMenu(false);
+
+<ContextualMenu
+  hidden={!showContextualMenu}
+  onItemClick={onHideContextualMenu}
+  onDismiss={onHideContextualMenu
+/>;
 
-### AntDesign DOM Structure
-
-_Note:_ Some long class names removed
-
-```HTML
-<ul role="menu">
-   <li class="ant-menu-item" role="menuitem" style="padding-left: 24px;">Item one</li>
-   <li class=" ant-menu-item-divider"></li>
-   <li class="ant-menu-item" role="menuitem" style="padding-left: 24px;"> Item two</li>
-   <li class="ant-menu-submenu ant-menu-submenu-inline" role="menuitem">
-      <div class="ant-menu-submenu-title" aria-expanded="false" aria-haspopup="true" style="padding-left: 24px;">
-          <span>Item Three</span>
-          <i class="ant-menu-submenu-arrow"/>
-      </div>
-   </li>
-</ul>
+// v0 uncontrolled Popup
+const [open, setOpen] = React.useState(false);
+
+<Popup
+  onOpenChange={(e, props: PopupProps) => {/*react on changes*/}}
+  trigger={<Button icon={<OpenOutsideIcon />} title="Open popup" />}
+/>;
+
+// v0 controlled Popup - used with trigger disables autocontrol
+const [open, setOpen] = React.useState(false);
+
+<Popup
+  open={open}
+  onOpenChange={(e, props: PopupProps) => setOpen(!props.open)}
+  trigger={<Button icon={<OpenOutsideIcon />} title="Open popup" />}
+/>;
 ```
 
-## Recommendations
-
-### Recommended DOM Structure
-
-```HTML
-<div role="menu">
-    <div role="menuitem">{"Item one"}</div>
-    <div role="presentation" />
-    <div role="menuitem">{"Item two"}</div>
-    <a role="menuitem" href="www.isALink.com"> link1 </a>
-    <div role="menuitem">
-        {"Item Three"}
-    </div>
-        <!-->
-            I'm not sure if this is the right role or if it is actually a menu item. It should
-            probably be related to its parent's item in some way.
-        <-->
-    <div role="presentation">
-        <!-->
-            Note: Does not necessarily need to be part of the same dom
-            tree, could be floating.
-        <-->
-        <div role="menu">
-            <div role="menuitem">{"SubItem one"}</div>
-            <div role="menuitem">{"SubItem two"}</div>
-        </div>
-    </div>
-</ul>
+### Keyboard navigation
+
+Both v7 and v0 support arrow navigation in the menu, and home/end keys to jump to first and last items respectively.
```

Comment:
there is also quite complex interaction with regards to left/right arrow keys related to opening and closing submenus described by ARIA which v0 implements.